x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/GetSearchApplicationAction.java

```diff
@@ -99,14 +128,46 @@ public Response(
             long updatedAtMillis,
             SearchApplicationTemplate template
         ) {
+            if (Strings.isNullOrEmpty(name)) {
+                throw new IllegalArgumentException("Search Application name cannot be null or blank");
+            }
+            // if (name!=null){
             this.searchApp = new SearchApplication(name, indices, analyticsCollectionName, updatedAtMillis, template);
+            // }
+            // else {
+            // this
+            // }
+
+        }
+
+        private static final ConstructingObjectParser<Response, String> PARSER = new ConstructingObjectParser<>(
+            "get_search_application_response",
+            p -> new Response((SearchApplication) p[0])
+        );
+        public static final ParseField SEARCH_APPLICATION_FIELD = new ParseField("searchApp");
+
+        static {
+            PARSER.declareObject(constructorArg(), (p, c) -> SearchApplication.fromXContent(c, p), SEARCH_APPLICATION_FIELD);
+        }
+
+        public static Response parse(XContentParser parser) {
+            return PARSER.apply(parser, null);
+        }
+
+        public static Response fromXContent(String resourceName, XContentParser parser) throws IOException {
+            // TODO this seems not good
```

Comment:
What is the problem here? 🤔

Reply:
I put that in, I wasn't sure if I should construct the object in `fromXContent`. I think it's OK though - but the comment should be removed.